test/plugin/scenarios/spring-scheduled-scenario/src/main/java/org/apache/skywalking/apm/testcase/spring/scheduled/job/SchedulingJob.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.testcase.spring.scheduled.job;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+@Component
+@EnableScheduling
+public class SchedulingJob {
+
+    private static final Logger logger = LogManager.getLogger(SchedulingJob.class);
+    
+    private static final OkHttpClient client = new OkHttpClient.Builder().build();
+
+    @Scheduled(fixedDelay = 5000)
```

Comment:
@hailin0, I think I got @zhaoyuguang 's point as he DM me.

This bean has `@Component` already, so, why this method wouldn't work? I think Spring annotation plugin tracing all methods of the bean. Unless you only want to activate this, rather than other beans?

Reply:
I found some bugs...  @wu-sheng 

Expected to be normal when I use @Configuration
-------------------

![9CFE35D62F59B825122F05C9D3555F27](https://user-images.githubusercontent.com/14371345/90636027-9a311d00-e25c-11ea-9e4f-1261fe2d2240.png)
![A37D3CE0897B9D84C08F71E841E6AF46](https://user-images.githubusercontent.com/14371345/90636076-aa48fc80-e25c-11ea-8f26-4a16c7a121d7.png)


Duplicate tracing occurs when I use @Component
-------------------
![0C5D9D790774A98490A8E3513A7EFCE8](https://user-images.githubusercontent.com/14371345/90636208-dc5a5e80-e25c-11ea-8b73-95368df28454.png)
![1A2F323E9BFF2FC94BD11ACCD690C27B](https://user-images.githubusercontent.com/14371345/90636226-e2e8d600-e25c-11ea-8772-0eb0c05399f8.png)

Reply:
Maybe the @Scheduled method should be filtered out in spring-annotation-plugi.

1. @Scheduled is not an optional plugin
2. spring-scheduler can be started through configuration files instead of just annotations

Reply:
The first one is easy to explain, we don't watch the configuration. The second one is not clear, why `Duplicate tracing occurs when I use @component`? Who will trace twice?

Reply:
Does this duplicated tracing cause by this new plugin with `@component`?

Reply:
Note, I know this plugin should be useful at some points, but I need to make it clear.

Reply:
> The first one is easy to explain, we don't watch the configuration. The second one is not clear, why `Duplicate tracing occurs when I use @component`? Who will trace twice?

spring-scheduled-annotation-plugin trace once
then spring-annotation-plugin(optional-plugin) trace it again

Reply:
OK, seems byte-buddy added some features. FYI @apache/skywalking-committers. It used to can't instrument a class twice, but this test proved it is not the case today.

Reply:
According to the official doc, https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Configuration.html

`@Configuration` should only be used for `@Bean`. Is there abuse? I am not working on Spring for a long time, could you explain more about the use case.

Reply:
> According to the official doc, https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Configuration.html
> 
> `@Configuration` should only be used for `@Bean`. Is there abuse? I am not working on Spring for a long time, could you explain more about the use case.

The document describes @Configuration and @Scheduled can be used together

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/scheduling/annotation/EnableScheduling.html

Reply:
>  @configuration and @scheduled can be used together

Yes, but with `component` is the right usage?

Reply:
> > @configuration and @scheduled can be used together
> 
> Yes, but with `component` is the right usage?

The document indicates that `@Component` and `@Scheduled` can also be used together

Reply:
> > > @configuration and @scheduled can be used together
> > 
> > 
> > Yes, but with `component` is the right usage?
> 
> The document indicates that `@Component` and `@Scheduled` can also be used together

Duplicate tracing occurs when `@Component` and `@Scheduled` are used together

Reply:
This is a tricky point, I think maybe you should consider put this plugin as optional too. And add a FAQ doc to describe these cases, and provide recommended solutions.

Reply:
OK，I'll try

Reply:
I need to set it's an optional plugin? spring-annotation-plugin is already in optional list

Reply:
Do you think there is a problem with using @Component and @Async generate two spans?

It seems that @Component, @Scheduled and @Component, @Async are similar scenarios

Reply:
> Do you think there is a problem with using @component and @async generate two spans?
> 
> It seems that `@Component`, `@Scheduled` and `@Component`, `@Async` are similar scenarios

Reply:
> I need to set it's an optional plugin? 

I mean this scheduled plugin.

> Maybe we should keep the status quo and use SpringScheduled as a operationName.

You could have two, but only `SpringScheduled` as an OP name(also will be an endpoint) is not a good idea, you could use `SpringScheduled-` + `full class name` + `method name` as a better operation name. Because from the analysis perspective you need a meaningful name.

Reply:
> You could have two, but only `SpringScheduled` as an OP name(also will be an endpoint) is not a good idea, you could use `SpringScheduled-` + `full class name` + `method name` as a better operation name. Because from the analysis perspective you need a meaningful name.

Yes, I agree with you.

So, are these the final conclusions?
1. spring-scheduled-annotation-plugin is not an optional plugin
2. use `SpringScheduled- + full class name + method name` as a better operation name.

Reply:
Yes, agree.